dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,127 @@
+ * system property. 
+ * 
+ * The {fat.bucket.db.type} property is set to different databases
+ * by our test infrastructure when a fat-suite has the property
+ * {fat.test.databases} set to true.
+ * 
+ * <br> Container Information: <br>
+ * DERBY: Uses a NOOP Container <br>
+ * DB2: Uses <a href="https://hub.docker.com/r/ibmcom/db2">Offical DB2 Container</a> <br>
+ * Oracle: TODO Update once we are using an official oracle container. <br>
+ * Postgre: Uses <a href="https://hub.docker.com/_/postgres">Offical Postgres Container</a> <br>
+ * MS SQL Server: Uses <a href="https://hub.docker.com/_/microsoft-mssql-server">Offical Microsoft SQL Container</a> <br>
+ * 
+ * TODO: Since this factory class is general it should be moved to fattest simplicity in the future
+ */
+public class DatabaseContainerFactory {
+	private static JdbcDatabaseContainer<?> container;
+	private static DatabaseContainerType dbContainerType;
+	
+	//Logging Constants
+	private static final Class<DatabaseContainerFactory> c = DatabaseContainerFactory.class;
+	
+	private DatabaseContainerFactory() {}
+	
+	/**
+	 * Reads the {fat.bucket.db.type} system property and returns a
+	 * container based on that property.
+	 * 
+	 * If we are using a network database, a test container running
+	 * that database will be returned.  
+	 * [Postgre, DB2, Oracle, SQLServer]
+	 * 
+	 * If we are using derby embedded or unsupported database
+	 * a no-op container will be returned.
+	 * [Derby, Informix, Sybase]
+	 * 
+	 * If {fat.bucket.db.type} is not set with a value, 
+	 * we will default to Derby Embedded.
+	 *
+	 * @return JdbcDatabaseContainer - The test container.
+	 */
+	public static JdbcDatabaseContainer<?> create() {
+		dbContainerType = DatabaseContainerType.getType(System.getProperty("fat.bucket.db.type", "Derby"));
+		container = initContainer();
+		return container;
+	}
+	
+	/**
+	 * Returns the DBContainerType being used for container creation.
+	 * @return DBContainerType
+	 */
+	public static DatabaseContainerType usingContainerType() {
+		if(dbContainerType != null) {
+			return dbContainerType;
+		} else {
+			throw new IllegalStateException("Container type is determined when creating a database container."
+					+ " Please call DatabaseContainerFactory.create before calling this method.");
+		}
+	}
+	
+	//Private Method: used to initialize test container.
+	private static JdbcDatabaseContainer<?> initContainer() {
+		switch(dbContainerType) {
+		case DB2:
+			return new Db2Container().acceptLicense()
+				  // Use 5m timeout for local runs, 15m timeout for remote runs
+				  .withStartupTimeout(Duration.ofMinutes(FATRunner.FAT_TEST_LOCALRUN ? 5 : 15))
+				  .withLogConsumer(DatabaseContainerFactory::log);
+		case ORACLE:
+			//Check if we are running local and unable to get JDBC driver, if so warn user.
+			if(FATRunner.FAT_TEST_LOCALRUN && !DatabaseContainerUtil.isJdbcDriverAvailable(dbContainerType)) {
```

Comment:
Instead of throwing an error we could also get rid of this check all together.  Then an error will be thrown from the test-containers code itself.  
```
[09/04/2019 10:08:54:858 CDT] 001 GenericContainer               tryStart                       S Could not start container
org.testcontainers.containers.JdbcDatabaseContainer$NoDriverFoundException: Could not get Driver
	at org.testcontainers.containers.JdbcDatabaseContainer.getJdbcDriverInstance(JdbcDatabaseContainer.java:169)
	at org.testcontainers.containers.JdbcDatabaseContainer.createConnection(JdbcDatabaseContainer.java:191)
	at org.testcontainers.containers.JdbcDatabaseContainer.waitUntilContainerStarted(JdbcDatabaseContainer.java:129)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:332)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:285)
```

Which seems pretty self explanatory